File: 0110-balanced-binary-tree/0110-balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool balanced = true;
    int solve(TreeNode * root){
        if(root == NULL) return 0;
        
        int left = solve(root -> left);
        int right = solve(root -> right);
        
        if(abs(left - right) > 1){
            balanced = false;
        }
        
        return max(left , right) + 1;
    }
    bool isBalanced(TreeNode* root) {
        solve(root);
        return balanced ;
    }
};

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    boolean balanced = true;
    int solve(TreeNode  root){
        if(root == null) return 0;
        
        int left = solve(root.left);
        int right = solve(root.right);
        
        if(Math.abs(left - right) > 1){
            this.balanced = false;
        }
        
        return Math.max(left , right) + 1;
    }
    public boolean isBalanced(TreeNode root) {
        solve(root);
        return balanced ;
    }
}